jwst/associations/mkpool.py

```diff
@@ -1,47 +1,223 @@
 """
 Tools for pool creation
 """
+import logging
+
 from astropy.io.fits import getheader as fits_getheader
-import numpy as np
 
 from . import AssociationPool
 
+__all__ = ['mkpool']
+
+# Configure logging
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+LogLevels = [logging.WARNING, logging.INFO, logging.DEBUG]
+
+# Header keywords to ignore
 IGNORE_KEYS = ('', 'COMMENT', 'HISTORY')
 
+# Non-header columns that need to be defined
+NON_HEADER_COLS = {
+    'asn_candidate': None,
+    'dms_note': '',
+    'is_imprt': 'f',
+    'is_psf': 'f',
+    'pntgtype': 'science',
+    'targetid': '1',
+}
+
+
+def mkpool(data,
+           asn_candidate=NON_HEADER_COLS['asn_candidate'], dms_note=NON_HEADER_COLS['dms_note'],
+           is_imprt=NON_HEADER_COLS['is_imprt'], is_psf=NON_HEADER_COLS['is_psf'],
+           pntgtype=NON_HEADER_COLS['pntgtype'],
+           **kwargs):
+    """Create an association pool from a list of FITS files.
+
+    Normally, association pools and the associations generated from those pools
+    are created by the automatic ground system process. Users should download
+    and modify those pools if need be. If desired, this function can be used to
+    create pools from scratch using a list of FITS files. Once created, the
+    :py:func:`~jwst.associations.generate` can be used to create associations
+    from these pools.
+
+    A number of columns used by the Association rules cannot be derived from the header
```

Comment:
perhaps insert "pool" ahead of "columns", just to make it painfully obvious what columns we're talking about

Reply:
✅